app/views/layouts/application.html.erb

```diff
@@ -32,7 +32,7 @@
           localStorage.removeItem('previous-search-url');
           window.location = searchUrl;
       }
-
+      window.SCP.userAccessTokenObject = <%= raw get_user_access_token(current_user).to_json %>
       window.SCP.userAccessToken = '<%= get_user_access_token(current_user) %>';
```

Comment:
This may well work!  However, if it does, it'd be through a surprising mechanism that would confuse maintainers.

I just pulled this down, and see that `window.SCP.userAccessTokenObject` is set to a string value (`"ya29...`) here, not JSON as the code assumes.  That's because this `get_user_access_token` returns a string, not JSON.  This causes `renewalTime` over in `setUpRenewalForUserAccessToken` to get set to `NaN` on the first evaluation, which triggers an immediate request to `/site/renew_user_access_token`.  That endpoint then returns JSON as expected, and seems to set things right.

So this seems like it'll cause `/site/renew_user_access_token` to always be called soon (but not immediately) on page load, which is unexpected.  

I see other calls to `/search/facets`, `/search?type=gene&page=1`, and `/search?type=study&page=1` and `/site/check_terra_tos_acceptance` all precede that.  Per my [notes yesterday](https://github.com/broadinstitute/single_cell_portal_core/pull/1845#discussion_r1282309732), that initial 4-call volley will use the value of `userAccessToken` as set in line 36 here.  I'm curious if `userAccessToken` as set in line 36 still gets initialized to an empty string now.  If so, then the initial call volley seems likely to return unexpected results, since they depend in part on a user's authentication / identity, e.g. search results partly depend on what you're authorized to see.

I realize this is a lot of detail; I note it here to hopefully help clarify complicated auth flows.  I'm happy to chat or pair if that seems useful!

Reply:
Yep definitely committed that too soon, please see: [a6eaa2c](https://github.com/broadinstitute/single_cell_portal_core/pull/1845/commits/a6eaa2cf3af35447792ea6c602c715deafd0dcc5) for better handling